src/main.cc

```diff
@@ -296,8 +318,653 @@ uint64_t rotr64 (uint64_t n, unsigned int c)
+                    way_fill = ooo_cpu[cpu].L2C.find_victim(cpu, instr_id, set, ooo_cpu[cpu].L2C.block[set], ip, pt2s, type);
+                    ooo_cpu[cpu].L2C.update_replacement_state(cpu, set, way_fill, pt2s, ip, ooo_cpu[cpu].L2C.block[set][way_fill].full_addr, type, 0);
+
+                    if (ooo_cpu[cpu].L2C.block[set][way_fill].valid == 0)
+                        ooo_cpu[cpu].L2C.block[set][way_fill].valid = 1;
+
+                    ooo_cpu[cpu].L2C.block[set][way_fill].dirty = 0;
+                    ooo_cpu[cpu].L2C.block[set][way_fill].prefetch = 0;
+                    ooo_cpu[cpu].L2C.block[set][way_fill].used = 0;
+
+                    ooo_cpu[cpu].L2C.block[set][way_fill].tag = pt2s;
+                    ooo_cpu[cpu].L2C.block[set][way_fill].address = pt2s;
+                    ooo_cpu[cpu].L2C.block[set][way_fill].full_addr = pt2s;
+                    ooo_cpu[cpu].L2C.block[set][way_fill].data = 55;// random value (not used)
+                    ooo_cpu[cpu].L2C.block[set][way_fill].cpu = cpu;
+
+                    // now search in LLC
+                    set = uncore.LLC.get_set(pt2s);
+                    way_read = uncore.LLC.check_hit(&search_packet);
+                    if(way_read >=0){
+                        // LLC read hit 
+                        uncore.LLC.llc_update_replacement_state(cpu, set, way_read, uncore.LLC.block[set][way_read].full_addr, ip, 0, type, 1);
+                        uncore.LLC.block[set][way_read].used = 1;
+                        cstall += 20;
+                    }
+                    else{
+                        // LLC read miss 
+                        cstall += 20;
+
+                        // fill LLC if you want to... this is a decision you have to make
+                        way_fill = uncore.LLC.llc_find_victim(cpu, instr_id, set, uncore.LLC.block[set], ip, pt2s, type);
+                        uncore.LLC.llc_update_replacement_state(cpu, set, way_fill, pt2s, ip, uncore.LLC.block[set][way_fill].full_addr, type, 0);
+
+                        if (uncore.LLC.block[set][way_fill].valid == 0)
+                            uncore.LLC.block[set][way_fill].valid = 1;
+
+                        uncore.LLC.block[set][way_fill].dirty = 0;
+                        uncore.LLC.block[set][way_fill].prefetch = 0;
+                        uncore.LLC.block[set][way_fill].used = 0;
+
+                        uncore.LLC.block[set][way_fill].tag = pt2s;
+                        uncore.LLC.block[set][way_fill].address = pt2s;
+                        uncore.LLC.block[set][way_fill].full_addr = pt2s;
+                        uncore.LLC.block[set][way_fill].data = 55;// random value (not used)
+                        uncore.LLC.block[set][way_fill].cpu = cpu;
+
+                        // DRAM access 
+                        cstall += 200;
+                    }
+                }
+            }
+        }
+    }
+
+    return cstall;
+}
+
```

Comment:
This seems to duplicate a lot of code that is in cache.cc. Is it wrong here to add the miss entries to the L1 read queues and allow the cache architecture to propogate the request?

My primary concern here is that all of these fills happen instantaneously, which is unrealistic. The LLC fill should take some time, and we should reflect that. However, all of these requests need to be filled _as a part of the STLB miss_. This is a difficult problem to solve. Could we set the STLB MSHR event_cycle to some absurdly large value, ensuring the core will wait on it until the requests from the page walk are resolved?

This section of code is the most important part of this contribution. We should be careful to get this right.